src/protocols/Http.ts

```diff
@@ -281,9 +292,27 @@ export default class HttpProtocol extends KuzzleAbstractProtocol {
       url += '?' + queryString.join('&');
     }
 
-    this._sendHttpRequest(method, url, payload)
-      .then(response => this.emit(payload.requestId, response))
-      .catch(error => this.emit(payload.requestId, {error}));
+    return {
+      method,
+      url,
+      payload
+    };
+  }
+
+  /**
+   * Sends a payload to the connected server
+   *
+   * @param {Object} data
+   * @returns {Promise<any>}
+   */
+  send (request: RequestPayload, options: JSONObject = {}) {
+    const verifiedRequest = this.formatRequest(request, options);
+
+    if (verifiedRequest) {
+      this._sendHttpRequest(verifiedRequest.method, verifiedRequest.url, verifiedRequest.payload)
```

Comment:
Nitpicking/boyscout: the `sendHttpRequest` could be refactored to take only the `verifiedRequest` object